cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/MonitorHandlerModule.java

```diff
@@ -178,4 +182,43 @@ public DatasetFramework get() {
       return new InMemoryDatasetFramework(new DefaultDatasetDefinitionRegistryFactory(injector), datasetModules);
     }
   }
+
+  /**
+   * Provides for {@link MasterServiceManager} used in non-hadoop environment.
+   */
+  private static final class NonHadoopMasterServiceManagerProvider implements Provider<MasterServiceManager> {
+
+    private final Class<? extends MasterServiceManager> serviceManagerClass;
+    private Injector injector;
+
+    NonHadoopMasterServiceManagerProvider(Class<? extends MasterServiceManager> serviceManagerClass) {
+      this.serviceManagerClass = serviceManagerClass;
+    }
+
+    @Inject
+    void setInjector(Injector injector) {
+      // Add a binding to TwillRunner if it doesn't exists. It is for unit-test and SDK environment.
```

Comment:
can we make this separate binding in a test setup?

Reply:
Done